### PR TITLE
untangled alertmanager config

### PR DESCRIPTION
**Pull request by richardtief**
Complete rewrite of the Alert Manager configuration. Separation of the individual inhibit rules and routes into dedicated Alertmanager configuration resources. Legacy Slack channels have been removed.

### CODE REVIEW

File: global/prometheus-alertmanager-operated/templates/am-config-route-octobus-mom.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+spec:
+  route:
+    receiver: octobus
+    matchers:
+      - name: severity
+        matchType: "=~"
+        value: "info|warning|critical"
+      - name: region
+        matchType: "=~"
+        value: "ap-ae-1|ap-au-1|ap-cn-1|ap-jp-1|ap-jp-2|ap-sa-1|ap-sa-2|eu-de-1|eu-de-2|eu-nl-1|la-br-1|na-ca-1|na-us-1|na-us-2|na-us-3"
```
Comment (viennaa) [inline global/prometheus-alertmanager-operated/templates/am-config-route-octobus-mom.yaml:16]:
```suggestion
        value: {{ include "regions" . }}
```

File: global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-keppel.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: AlertmanagerConfig
+metadata:
+  name: inhibit-rules-keppel
+  labels:
+    alertmanager: {{ include "alertmanagerRelease.name" . }}
+spec:
+  inhibitRules:
+{{- range tuple "ap-ae-1" "ap-au-1" "ap-cn-1" "ap-jp-1" "ap-jp-2" "ap-sa-1" "ap-sa-2" "eu-de-1" "eu-de-2" "eu-nl-1" "la-br-1" "na-ca-1" "na-us-1" "na-us-2" "na-us-3" "qa-de-1" }}
```
Comment (viennaa) [inline global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-keppel.yaml:9]: use the withQADE1 helper

**Review by viennaa — CHANGES_REQUESTED** [review]
Thanks for doing the heavy lifting, some small nits inline.

File: global/prometheus-alertmanager-operated/templates/_alertmanager.yaml.tpl [hunk 3]
```diff
@@ -1,1048 +1,17 @@
...
-        text: {{"'{{ .CommonAnnotations.mail_body }}'"}}
-        html: {{"'{{ template \"cc_email_receiver.KubernikusKlusterLowOnObjectStoreQuota\" . }}'"}}
-        smarthost: {{ required ".Values.cc_email_receiver.smtp_host undefined" .Values.cc_email_receiver.smtp_host | quote }}
-        auth_username: {{ required ".Values.cc_email_receiver.auth_username undefined" .Values.cc_email_receiver.auth_username | quote }}
-        auth_password: {{ required ".Values.cc_email_receiver.auth_password undefined" .Values.cc_email_receiver.auth_password | quote }}
-        require_tls: true
-        send_resolved: false
-  {{- end }}
+templates:
+  - /notification-templates/*.tmpl
```
[the remark this answers is not in the document]
Comment (richardtief) [reply global/prometheus-alertmanager-operated/templates/_alertmanager.yaml.tpl:17]: The templates are mounted in the pod, which contains the Slack, e-mail, etc. templates. So they do exist
 
https://github.com/sapcc/helm-charts/blob/ca45a0fad816db6ac90119eb83c6b9f63031cc3d/common/prometheus-alertmanager-base/templates/alertmanager.yaml#L60-L72

File: global/prometheus-alertmanager-operated/templates/am-config-route-octobus-mom.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: AlertmanagerConfig
+metadata:
+  name: alertmanager-{{ include "alertmanagerRelease.name" . }}-route-octobus-mom
```
[the remark this answers is not in the document]
Comment (richardtief) [reply global/prometheus-alertmanager-operated/templates/am-config-route-octobus-mom.yaml:4]: I am open to change it. Will do it for every resource then.

File: global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-keppel.yaml (hunk 2 above)
Comment (richardtief) [reply global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-keppel.yaml:9]: The regions template is not a list. Will but it in such to make it reusable.

**Review by viennaa — APPROVED** [review]
Thanks, appreciated a lot!

**Pull request merged**